### DISCUSSION TITLE
Is web platform supported?

### DISCUSSION THREAD

**Discussion by munrocket** [Wumpf/wgpu-profiler#48] [Q&A] [answered]
Hello, I not so familiar with Rust and looking for profiler for compute.toys. Can I get timestamps for each compute stage?

**Comment by Wumpf** [comment, marked as answer]
~I don't quite understand how you'd use a Rust crate on the compute.toys website?~

Anyways, for the more general question whether this crate works with WebGPU: "Yes, it should, _but_"
The big issue is that right now you need to change a browser setting to enable timestamps. According to [this website](https://omar-shehata.medium.com/how-to-use-webgpu-timestamp-query-9bf81fb5344a) you need to start Chrome with `--disable-dawn-features=disallow_unsafe_apis`. The WebGPU spec generally does not require the timestamp feature to be available, so there is no general guarantee that timings can be extracted.

**Reply by munrocket** [reply]
Ok, this is great, thank you, I will try!

**Reply by munrocket** [reply]
We using Rust with Wasm there https://github.com/compute-toys/wgpu-compute-toy

**Reply by Wumpf** [reply]
ohh you're the _author_ of compute-toy, I missunderstood 😅

**Reply by Wumpf** [reply]
good luck, curious how it goes :)

**Reply by munrocket** [reply]
No no, just contributor.
